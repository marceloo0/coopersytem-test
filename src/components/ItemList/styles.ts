import styled, {css} from 'styled-components/native';

interface ContainerProps {
  disabled: boolean;
}

export const Container = styled.TouchableOpacity<ContainerProps>`
  background-color: #fff;
  margin-top: 3px;
  padding: 15px 20px;

  ${(props) =>
    props.disabled &&
    css`
      opacity: 0.5;
    `}
`;
export const Info = styled.View`
  flex-direction: row;
  justify-content: space-between;
`;
export const Title = styled.Text`
  font-size: 21px;
  font-weight: bold;
`;
export const Description = styled.Text`
  color: #c9c9c9;
  font-size: 16px;
`;
export const Value = styled.Text`
  font-size: 21px;
`;
