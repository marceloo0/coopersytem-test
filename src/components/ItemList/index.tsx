import React from 'react';

import {Container, Info, Title, Description, Value} from './styles';

interface ItemListProps {
  onPress(): void;
  disabled: boolean;
  title: string;
  description: string;
  value: string;
}

const ItemList: React.FC<ItemListProps> = ({
  onPress,
  disabled,
  title,
  description,
  value,
}) => {
  return (
    <Container onPress={onPress} disabled={disabled}>
      <Info>
        <Title>{title}</Title>
        <Value>{value}</Value>
      </Info>
      <Description>{description}</Description>
    </Container>
  );
};

export default ItemList;
