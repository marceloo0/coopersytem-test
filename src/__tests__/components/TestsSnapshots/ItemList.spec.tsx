import React from 'react';
import renderer from 'react-test-renderer';
import ItemList from '../../../components/ItemList';

jest.useFakeTimers();

test('renders correctly', () => {
  const itemList = renderer
    .create(
      <ItemList
        onPress={() => {}}
        disabled={true}
        description="Viajem dos sonhos"
        title="Investimento II"
        value="R$ 7.500,00"
      />,
    )
    .toJSON();
  expect(itemList).toMatchSnapshot();
});
