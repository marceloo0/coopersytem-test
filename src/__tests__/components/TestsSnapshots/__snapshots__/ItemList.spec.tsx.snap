// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableHighlight
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "marginTop": 3,
        "opacity": 0.5,
        "paddingBottom": 15,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 15,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 21,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Investimento II
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 21,
          },
        ]
      }
    >
      R$ 7.500,00
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#c9c9c9",
          "fontSize": 16,
        },
      ]
    }
  >
    Viajem dos sonhos
  </Text>
</TouchableHighlight>
`;
